shap/utils/_types.py

```diff
@@ -0,0 +1,8 @@
+from typing import TypeAlias, TypeVar, Union
+
+import numpy as np
+import pandas as pd
+import scipy.sparse
+
+_ArrayLike: TypeAlias = Union[np.ndarray, pd.DataFrame, pd.Series, list, scipy.spare.csr_matrix]
+_ArrayT = TypeVar("_ArrayT", np.ndarray, pd.DataFrame, pd.Series, scipy.sparse.csr_matrix, list)
```

Comment:
I thought it would be useful to have a new submodule for common type aliases.

Just to draw attention to something I found a bit subtle: the `Union` and `TypeVar` are very similar, but the latter was useful to annotate that a function will return exactly the same type as the input type.

More info: https://stackoverflow.com/a/58903907

Example use in `_general.py : sample()`